Code/GraphMol/FileParsers/GeneralFileReader.h

```diff
@@ -0,0 +1,213 @@
+#ifndef GENERAL_FILE_READER_H
+#define GENERAL_FILE_READER_H
+#include <string>
+#include <iostream>
+#include <vector>
+#include <boost/filesystem.hpp>
+#include "MolSupplier.h"
+#include <RDGeneral/BadFileException.h>
+#include <RDStreams/streams.h>
+
+namespace fs = boost::filesystem;
+
+namespace RDKit{
+	struct SupplierOption{
+		bool takeOwnership = true;
+		bool sanitize = true;
+		bool removeHs = true;
+		bool strictParsing = true;
+
+		std::string delimiter = "\t";
+		int smilesColumn = 0;
+		int nameColumn = 1;
+		bool titleLine = true;
+
+		std::string nameRecord = "";
+		int confId2D = -1;
+		int confId3D = 0;
+	};
+
+  class GeneralFileReader{
+	  public:
+			std::string fname;
+		  std::string file_format, compression_format;
+
+		struct SupplierOption opt;	// options for the Mol Supplier 
+		public:
+			GeneralFileReader(std::string fileName);
+		  GeneralFileReader(std::string fileName, struct SupplierOption options);
+
+		  /*
```

Comment:
Again, for consistency please try and structure and format your comments like other RDKit header files.
`Atom.h` is a pretty good example of what this should look like: https://github.com/rdkit/rdkit/blob/master/Code/GraphMol/Atom.h